LPD-35070

SelenaAungst opened the pull request:
https://liferay.atlassian.net/browse/LPD-35070

## Goal
To provide the theme selection when creating a style book

## Approach
Retrieve the list of available frontend token definitions and their theme ids.

## Limitation
This task does not aim to show the frontend tokens available for the chosen theme. That feature will be implemented at https://liferay.atlassian.net/browse/LPD-35071


Resent from https://github.com/liferay-platform-experience/liferay-portal/pull/789

cc @thiago-buarqque @ethib137 @algobob

liferay-continuous-integration commented:
The following guidelines have been set by the owner of this repository: 
- &nbsp;&nbsp;&nbsp;&nbsp;"Thank you for sending this PR to liferay-platform-experience. If you haven't already,take this opportunity to review our [pull request guidelines](https://liferay.atlassian.net/wiki/spaces/PEDS/pages/2915598365). For additional help,please reach out via the [appropriate channels](https://liferay.atlassian.net/wiki/spaces/PEDS/overview#%3Aslack%3A-Slack-Channels)."

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

SelenaAungst commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-40.liferay.com/job/test-portal-source-format/6690/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6690/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/28a1eeabb17c71edd1522aae520f5c153a13bf83">28a1eeabb17c71edd1522aae520f5c153a13bf83</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/SelenaAungst/liferay-portal/tree/LPD-35070-5">LPD-35070-5</a><br/>Branch GIT ID: <a href="https://github.com/SelenaAungst/liferay-portal/commit/7f2ca4a27b15d7306bcf19da178c5a5b3d3a921c">7f2ca4a27b15d7306bcf19da178c5a5b3d3a921c</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-40.liferay.com/job/test-portal-source-format/6690/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6690/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-40.liferay.com/job/test-portal-source-format/6690/">test-portal-source-format#6690</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/6690/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-platform-experience/liferay-portal/pull/791">liferay-platform-experience#791</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/69270897">[master] ci:test:sf - SelenaAungst &gt; liferay-platform-experience - PR#791 - 2024-11-14[17:39:54]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>69270897<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-40.liferay.com/job/publish-testray-report/25585/">publish-testray-report#25585</a><br/></div></div>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/11818/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11818/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:stable - 24 out of 24 jobs passed</h3><h3>:x: ci:test:relevant - 28 out of 30 jobs passed in 1 hour 43 minutes</h3><details><summary>Click here for more details.</summary><br/><p>This pull is eligible for <a href="https://grow.liferay.com/share/CI+liferay-continuous-integration+GitHub+Commands#General-Commands">reevaluation</a>. When this <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)/1019">upstream build</a> has completed, using the following CI command will compare this pull request result against a more recent upstream result:<pre>ci:reevaluate:1397401_11818</pre></p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/28a1eeabb17c71edd1522aae520f5c153a13bf83">28a1eeabb17c71edd1522aae520f5c153a13bf83</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/35e18f698fa67ad7f1894d29ca9780e5345aaeba">35e18f698fa67ad7f1894d29ca9780e5345aaeba</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/#/project/35392/routines/590307/build/69128863">EE Development Acceptance (master) - 1018 - 2024-11-14[12:41:20]</a></p><details><summary><strong>ci:test:stable - 24 out of 24 jobs PASSED</strong></summary><details><summary><strong>24 Successful Jobs:</strong></summary><ul/></details></details><details><summary><strong>ci:test:relevant - 28 out of 30 jobs PASSED</strong></summary><div><h4>2 Failed Jobs:</h4><ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/11818/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>central-requirements<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785178/">central-requirements/0/0</a></li></ul></li><li>functional-smoke-tomcat90-mysql57_stable<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785188/">functional-smoke-tomcat90-mysql57_stable/0/0</a></li></ul></li><li>functional-upgrade-tomcat90-mysql57_stable<ul><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1256285/">functional-upgrade-tomcat90-mysql57_stable/0/0</a></li><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785185/">functional-upgrade-tomcat90-mysql57_stable/0/1</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022861/">functional-upgrade-tomcat90-mysql57_stable/0/2</a></li></ul></li><li>js-unit<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785179/">js-unit/0/0</a></li><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785180/">js-unit/1/0</a></li></ul></li><li>modules-compile<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022862/">modules-compile/0/0</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/718206/">modules-compile/0/1</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142135/">modules-compile/0/2</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1137742/">modules-compile/0/3</a></li><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785190/">modules-compile/0/4</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022860/">modules-compile/0/5</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/718204/">modules-compile/0/6</a></li><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1137740/">modules-compile/0/7</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142137/">modules-compile/0/8</a></li><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1256284/">modules-compile/0/9</a></li><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142136/">modules-compile/0/10</a></li><li><a href="https://test-1-25.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/718205/">modules-compile/0/11</a></li><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1028600/">modules-compile/0/12</a></li></ul></li><li>modules-integration-mysql57<ul><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170/"><strike>modules-integration-mysql57/0/0</strike></a></li></ul></li><li>modules-integration-mysql57_stable<ul><li><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1028601/">modules-integration-mysql57_stable/0/0</a></li></ul></li><li>modules-semantic-versioning<ul><li><a href="https://test-1-27.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1125380/">modules-semantic-versioning/0/0</a></li></ul></li><li>modules-unit<ul><li><a href="https://test-1-32.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1137741/">modules-unit/0/0</a></li></ul></li><li>modules-unit_stable<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022859/">modules-unit_stable/0/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785187/">playwright-js-tomcat90-mysql57/0/0</a></li><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022858/">playwright-js-tomcat90-mysql57/1/0</a></li></ul></li><li>playwright-js-tomcat90-mysql57_stable<ul><li><a href="https://test-1-40.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/785186/">playwright-js-tomcat90-mysql57_stable/0/0</a></li></ul></li><li>unit_stable<ul><li><a href="https://test-1-28.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1022857/">unit_stable/0/0</a></li></ul></li></ul></div><details><summary><strong>28 Successful Jobs:</strong></summary><ul/></details></details><h5>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11818/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170//consoleText">modules-integration-mysql57/0/0</a><ol><li><details><summary><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170/testReport/com.liferay.site.initializer.extender.internal.test/BundleSiteInitializerTest">com.liferay.site.initializer.extender.internal.test.BundleSiteInitializerTest</a> - <a href="https://testray.liferay.com/home/-/testray/case_results/69033116/history">Failed 7 of last 24</a></summary><div><ol><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170//testReport/com.liferay.site.initializer.extender.internal.test/BundleSiteInitializerTest/testInitializeFromBundle">testInitializeFromBundle</a></li></ol></div></details></li></ol></div></li></ol><hr/><details><summary><strong>Failures in common with <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">acceptance upstream results</a> at 35e18f698fa67ad7f1894d29ca9780e5345aaeba:</strong></summary><ol><li><div><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170//consoleText">modules-integration-mysql57/0/0</a><ol><li><details><summary><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170/testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest">com.liferay.portal.log.assertor.PortalLogAssertorTest</a></summary><div><ol><li><a href="https://test-1-31.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/1142170//testReport/com.liferay.portal.log.assertor/PortalLogAssertorTest/testScanXMLLog">testScanXMLLog</a></li></ol></div></details></li></ol></div></li></ol></details></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11818/bundles/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a></li></ul>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-39.liferay.com/job/test-portal-acceptance-pullrequest(master)/11818/">test-portal-acceptance-pullrequest(master)#11818</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/11818/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>relevant<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-platform-experience/liferay-portal/pull/791">liferay-platform-experience#791</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/#/project/35392/routines/45357/build/69312727">[master] ci:test:relevant - SelenaAungst &gt; liferay-platform-experience - PR#791 - 2024-11-14[17:57:36]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>69312727<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-39.liferay.com/job/publish-testray-report/25487/">publish-testray-report#25487</a><br/></div></div>

pat270 commented:
@SelenaAungst one thing I noticed is that when I create a new Style Book based on Dialect is that the Dialect tokens don't seem to be applying. I added a gif to explain better.

![dialect](https://github.com/user-attachments/assets/042778fd-b7f1-46cd-876d-a75513fc408f)

SelenaAungst commented:
I think this what the limitation is in the description of this PR, if so it should be handled in the linked ticket. Please let me know if I am mistaken and I can look into this
@pat270

thiago-buarqque commented:
@pat270 @SelenaAungst That's it, it's something for the next task. This info is in the PR description. The idea is to keep PRs short

thiago-buarqque reviewed (approved):
Backend side LGTM. 

@algobob what do you think?

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/StyleBookModal.js:
```diff
@@ -0,0 +1,216 @@
+	const handleSubmit = (event) => {
+		event.preventDefault();
+
+		const errors = validateFields(name, frontendTokenDefinition);
```
In general, compare your implementation against the simple input modal. this is a better example to go from.

SelenaAungst commented:
I marked most of the review comments with a thumbs up, those I added in a commit just now. The rest I ran into issues with while testing, so I will continue working on these tomorrow.

SelenaAungst commented:
> @SelenaAungst reminder: Feature flag is already in master =)

@algobob Ah, so should I remove the feature flag from here?

algobob commented:
> @algobob Ah, so should I remove the feature flag from here?

oh no.  it was just a reminder that flag is in portal.properties in master already :slightly_smiling_face:

SelenaAungst commented:
Currently making the changes requested in: https://github.com/SelenaAungst/liferay-portal/pull/5#issuecomment-2517480213

- Typescript changes
- Modal changes
- Switch to Clay Picker instead of Clay Select

SelenaAungst commented:
Hi @ethib137,

I made all the requested changes. I ended up combining your suggestions with some of the changes I already made, please let me know if they look good or if I should just use all of the changes from your PR. 

Additionally, the change to Picker works currently, but for some reason you need to click on Dialect twice before it can be selected. I think this might be related to onSelectionChange, I tried using the same code from onChange there instead but ran into issues. I will continue working on that tomorrow, please let me know if you have any additional suggestions for this.

Thank you!

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+					</Picker>
+				</FieldBase>
+
+				<ClayForm onSubmit={handleSubmit} ref={formRef}>
```
Onsubmit isn't doing anything right now because the submit button isn't associated with this form. We can have the submit button submit the form even though it's not a child of the form by adding the form attribute to the button pointing to the id of the form.

So we should add an id on this form.

ethib137 replied:
Also, the form should wrap both inputs.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+
+						<ClayButton
+							displayType="primary"
+							onClick={handleSubmit}
```
instead of adding an onClick, we should add the [form attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#form) pointing to the id of the form. We should also add the attribute `type="submit"`.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+						'the-style-book-will-be-created-based-on-the-selected-token-definition'
+					)}
+					id={themeIdId}
+					label={Liferay.Language.get('create-style-book-for')}
```
For both inputs we should be using FieldBase's errorMessage prop to show the error message related to the field directly on the field.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0]);
+
+	const formRef = useRef(null);
```
I don't think  we need the formRef.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+	const handleSubmit = (event: any) => {
+		event.preventDefault();
+
+		const error = name.trim().length
```
The logic for checking whether the name is added is being duplicated in multiple locations. This should be made it's own shared method.

Additionally we need to show this error directly on the field not above both fields.

ethib137 replied:
To share this logic can create function:

```
validateName(name) {
    let error = '';

    if (name.trim().length > 0) {
        error = Liferay.Language.get('this-field-is-required');
    }

    setErrorMessage(error);

    return error;
}
```

This function will need to be inside the React component since it is calling setErrorMessage which is a useState method and requires the component context.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/java/com/liferay/style/book/web/internal/display/context/StyleBookManagementToolbarDisplayContext.java:
```diff
@@ -126,6 +144,46 @@ public Map<String, Object> getAdditionalProps() {
+
+					frontendTokenDefinitionProviders.add(
+						HashMapBuilder.<String, Object>put(
+							"name", name
```
The name shouldn't just be the theme or CX name. It needs to be suffixed by it's type. So it should be either `x-theme` or `x-theme-css-client-extension` depending on what it is. This is important because we could have a theme and a theme css client extension with the same name.

SelenaAungst replied:
Just to confirm, should these language keys be the following:
x-theme={0} Theme
x-theme-css-client-extension={0} Theme CSS Client Extension

I referred to this https://github.com/brianchandotcom/liferay-portal/pull/58629#issuecomment-390004381 and labeled them as titles

SelenaAungst commented:
Thanks @ethib137, I'll make these changes today

SelenaAungst commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+	namespace,
+}: AddStyleBookModalProps) =>{
+
+	const [active, setActive] = useState<boolean>(false);
```
Ah that makes sense, thank you!

SelenaAungst commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,212 @@
+							errorMessage ? 'has-error' : ''
+						}`}
+					>
+						<label
```
Should FieldBase replace the div or label here?

SelenaAungst commented:
I added most of these changes. What I tried with the added name variables was not working, so I am still working on that part and anything from your suggestions above that I did not put a thumbs up on.

SelenaAungst commented:
Hi @ethib137,

I added everything here except for consolidating the the name check logic, I tried multiple ways of this but for some reason I could not get it to update properly when I tried to move things. Additionally, I added the id to the ClayForm for the form attribute, I was a bit confused on that part but hopefully this is in line with what you meant. 

Please let me know if any additional changes are needed here, if no I can send all of this in a new PR with correct commit names and combine these changes.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,192 @@
+			<ClayModal.Body>
+
+				<ClayForm id={saveButtonId} onSubmit={handleSubmit}>
+
```
No new line here.

ethib137 commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,192 @@
+						>
+							{Liferay.Language.get('name')}
+
+							<span className="reference-mark">
```
This should be handled by fieldbase. It accepts a parameter required.

ethib137 reviewed (changes requested):
Nice Work @SelenaAungst

matuzalemsteles commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,192 @@
+						id={themeIdId}
+						label={Liferay.Language.get('create-style-book-for')}
+					>
+						<Picker
+							defaultSelectedKey={themeId}
+							id={themeIdId}
+							onSelectionChange={setThemeId}
+							selectedKey={themeId.themeId}
+						>
+							{frontendTokenDefinitionProviders.map(
+								(frontendTokenDefinitionProvider) => (
+									<Option
+										key={frontendTokenDefinitionProvider.themeId}
+										textValue={frontendTokenDefinitionProvider.name}
+									>
+										{frontendTokenDefinitionProvider.name}
+									</Option>
+								)
+							)}
+						</Picker>
```
Instead of rendering the options in a map directly as a child of the Picker, it would be better to pass the items to the Picker so that you can have more benefits from the Picker being able to control and cache when necessary for optimizations and adjustments on mobile as well. This makes the component dynamically rendered instead of static.

```jsx
<Picker
	defaultSelectedKey={themeId}
	id={themeIdId}
	onSelectionChange={setThemeId}
	selectedKey={themeId.themeId}
	items={frontendTokenDefinitionProviders}
>
    {(item) => (
        <Option
           	key={item.themeId}
           	textValue={item.name}
        >
       	    {item.name}
        </Option>
    )}
</Picker>
```

matuzalemsteles commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,192 @@
+	const [themeId, setThemeId] =
+		useState<any>(frontendTokenDefinitionProviders[0].themeId);
+
+	const handleFormError = (responseContent: any) => {
```
I think it would be nice to add types instead of any.

matuzalemsteles commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,192 @@
+		setErrorMessage(responseContent.error || '');
+	};
+
+	const handleSubmit = (event: any) => {
```
Same here. I think we can use this `event: React.FormEvent<HTMLFormElement>`.

matuzalemsteles reviewed (commented):
I added some comments.

SelenaAungst commented:
@ethib137
The name change and removed blank lines have already been added, I can see it in the files changed section on the PR

SelenaAungst commented:
@ethib137 @matuzalemsteles 
I added all the changes you requested. 

I still run into the issue I had before with the validation change. When you click the save button, the message says "Please enter a valid name." from handleFormError, then if you type in text after that it says "This field is required." even though there is text in the field and it can be saved. I think this needs to be reset somehow but I am not sure.

thiago-buarqque commented:
@SelenaAungst the issue is this [condition here](https://github.com/liferay-platform-experience/liferay-portal/pull/791/files#diff-a4fa624ee85d882fedb2b0b7c6c2558eabd9b2eb54ff3e7bc7837fd12d9fab50R47), it should be the opposite

```typescript
if (!name.trim().length) {
  error = Liferay.Language.get('this-field-is-required');
}
```
instead of 

```typescript
if (name.trim().length) {
  error = Liferay.Language.get('this-field-is-required');
}
```

thiago-buarqque commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,178 @@
+							{Liferay.Language.get('cancel')}
+						</ClayButton>
+
+						<ClayButton
```
I'd enable the submit button only if the `name` is not empty.

thiago-buarqque reviewed (commented):
Thanks @SelenaAungst :D 

I am not frontend, but I think we can enhance the error handling in `AddStyleBookModalContent`. But please, refer to Evan and Matu for the official review.

SelenaAungst commented:
> @SelenaAungst the issue is this [condition here](https://github.com/liferay-platform-experience/liferay-portal/pull/791/files#diff-a4fa624ee85d882fedb2b0b7c6c2558eabd9b2eb54ff3e7bc7837fd12d9fab50R47), it should be the opposite

Oh wow oops... thank you!!

SelenaAungst commented:
@thiago-buarqque I made those changes, let me know if this looks better!

matuzalemsteles commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,183 @@
+
+		setErrorMessage(error);
+
+		setShowSaveButton(error ? true : false);
```
I think this would not be necessary to have another state to deal with disabling the button, you can use the error state itself to do this with `Boolean(error)`. This is good because we avoid a double re-render here, in the new versions in React this would not be a problem anymore because it is possible in batch but I believe it is unnecessary for this use case.

matuzalemsteles commented on modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/AddStyleBookModalContent.tsx:
```diff
@@ -0,0 +1,183 @@
+	};
+
+	const validateName = (name: string) => {
+		let error = '';
+
+		if (!name.trim().length) {
+			error = Liferay.Language.get('this-field-is-required');
+		}
+
+		setErrorMessage(error);
```
I think we can simplify this here by just adding: `setErrorMessage(!name.trim() ? Liferay.Language.get('this-field-is-required') : '')`;

matuzalemsteles reviewed (commented):
Well I added some comments but I believe this is not a blocker but just suggestions for "code quality".

SelenaAungst commented:
Thanks @matuzalemsteles, I pushed these changes

SelenaAungst commented:
reorganized and resent here: https://github.com/liferay-platform-experience/liferay-portal/pull/853
